python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py

```diff
@@ -553,117 +381,52 @@ def get_conn(self):
-                            )
-
             raw_res = super(DagsterOperator, self).execute(context)
             self.log.info('Finished executing container.')
-            res = None
-
-            # FIXME
-            # Unfortunately, log lines don't necessarily come back in order...
-            # This is error-prone, if something else logs JSON
-            lines = list(reversed(raw_res.decode('utf-8').split('\n')))
-            last_line = lines[0]
-
-            for line in lines:
-                try:
-                    res = json.loads(line)
-                    break
-                # If we don't get a GraphQL response, check the next line
-                except JSONDecodeError:
-                    continue
-
-            if res is None:
-                raise AirflowException('Unhandled error type. Response: {}'.format(last_line))
-
-            if res.get('errors'):
-                raise AirflowException(
-                    'Internal error in GraphQL request. Response: {}'.format(res)
-                )
+            (res, last_line) = parse_raw_res(raw_res)
+
+            handle_errors(res, last_line)
 
-            res_type = res['data']['startSubplanExecution']['__typename']
+            res_data = res['data']['executePlan']
+
+            res_type = res_data['__typename']
 
             if res_type == 'PipelineConfigValidationInvalid':
-                errors = [err['message'] for err in res['data']['startSubplanExecution']['errors']]
+                errors = [err['message'] for err in res_data['errors']]
                 raise AirflowException(
                     'Pipeline configuration invalid:\n{errors}'.format(errors='\n'.join(errors))
                 )
 
-            if res_type == 'StartSubplanExecutionSuccess':
-                self.log.info('Subplan execution succeeded.')
-                if res['data']['startSubplanExecution']['hasFailures']:
+            if res_type == 'PipelineNotFoundError':
+                raise AirflowException(
+                    'Pipeline {pipeline_name} not found: {message}:\n{stack_entries}'.format(
+                        pipeline_name=res_data['pipelineName'],
+                        message=res_data['message'],
+                        stack_entries='\n'.join(res_data['stack']),
+                    )
+                )
+
+            if res_type == 'ExecutePlanSuccess':
+                self.log.info('Plan execution succeeded.')
+                if res_data['hasFailures']:
                     errors = [
                         step['errorMessage']
-                        for step in res['data']['startSubplanExecution']['stepEvents']
+                        for step in res_data['stepEvents']
                         if not step['success']
                     ]
                     raise AirflowException(
                         'Subplan execution failed:\n{errors}'.format(errors='\n'.join(errors))
                     )
```

Comment:
I think we should either check an invariant if we don't recognize the return GraphQL type or throw a generic Airflow Exception

Reply:
that stuff happens on line 390 in `handle_errors`